contracts/factories/DelegatedManagerFactory.sol

```diff
@@ -0,0 +1,378 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity 0.6.10;
+pragma experimental ABIEncoderV2;
+
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { ISetToken } from "@setprotocol/set-protocol-v2/contracts/interfaces/ISetToken.sol";
+
+import { AddressArrayUtils } from "../lib/AddressArrayUtils.sol";
+import { DelegatedManager } from "../manager/DelegatedManager.sol";
+import { IDelegatedManager } from "../interfaces/IDelegatedManager.sol";
+import { ISetTokenCreator } from "../interfaces/ISetTokenCreator.sol";
+
+/**
+ * @title DelegatedManagerFactory
+ * @author Set Protocol
+ *
+ * Factory smart contract which gives asset managers the ability to:
+ * > create a Set Token managed with a DelegatedManager contract
+ * > create a DelegatedManager contract for an existing Set Token to migrate to
+ * > initialize extensions and modules for SetTokens using the DelegatedManager system
+ */
+contract DelegatedManagerFactory {
+    using AddressArrayUtils for address[];
+    using Address for address;
+
+    /* ============ Structs ============ */
+
+    struct InitializeParams{
+        address deployer;
+        address owner;
+        IDelegatedManager manager;
+        bool isPending;
+    }
+
+    /* ============ Events ============ */
+
+    /**
+     * @dev Emitted on DelegatedManager creation
+     * @param _setToken             Instance of the SetToken being levered
+     * @param _manager              Address of the DelegatedManager
+     * @param _deployer             Address of the deployer
+    */
+    event DelegatedManagerCreated(
+        ISetToken indexed _setToken,
+        DelegatedManager indexed _manager,
+        address _deployer
+    );
+
+    /**
+     * @dev Emitted on DelegatedManager initialization
+     * @param _setToken             Instance of the SetToken being levered
+     * @param _manager              Address of the DelegatedManager owner
+    */
+    event DelegatedManagerInitialized(
+        ISetToken indexed _setToken,
+        IDelegatedManager indexed _manager
+    );
+
+    /* ============ State Variables ============ */
+
+    // SetTokenFactory address
+    ISetTokenCreator public setTokenFactory;
+
+    // Mapping which stores manager creation metadata between creation and initialization steps
+    mapping(ISetToken=>InitializeParams) public initializeState;
+
+    // Mapping of all sets that have succesfully initialized
+    mapping(ISetToken=>bool) public isValidSet;
+
+    // Address array of all sets that have succesfully initialized
```

Comment:
Have reworded this in [e90f065](https://github.com/SetProtocol/set-v2-strategies/pull/14/commits/e90f0658086a2d3cf9ee2ead67f1224231a7db60) 

Maybe we should call this `isFactoryIssuedSet`? I can see how it might be convenient to query that on the front end but will ask Dylan about this.

Reply:
Resolving. This stuff was removed in [7e7e116](https://github.com/SetProtocol/set-v2-strategies/pull/14/commits/7e7e116f5492d58331c053460994d35247b28e9d). See comment below....